### PR TITLE
Add CycloneDX predicate

### PR DESCRIPTION
**Pull request by puerco**
**Fixes issue:**

None

**Description:**

This PR adds CycloneDX as a recognized predicate type 

Signed-off-by: Adolfo García Veytia (Puerco) <puerco@chainguard.dev>

**Please verify and check that the pull request fulfills the following
requirements:**

- [x] Tests have been added for the bug fix or new feature (NA)
- [x] Docs have been added for the bug fix or new feature (NA)

### CODE REVIEW

File: in_toto/model.go [hunk 1]
```diff
@@ -79,6 +79,8 @@ const (
 	// The SPDX mandates 'spdxVersion' field, so predicate type can omit
 	// version.
 	PredicateSPDX = "https://spdx.dev/Document"
+	// PredicateCycloneDX represents a CycloneDX SBOM
+	PredicateCycloneDX = "https://cyclonedx.org/schema"
```
Comment (jdolitsky) [inline in_toto/model.go:83]: Reason for `/schema` vs. `/Document` (as in SPDX version)?

Comment (puerco) [reply in_toto/model.go:83]: I think Document in spdx tries to refer to a SPDX document somehow. I pointed the URL to where the CycloneDX schemas live to make it more in line with the other predicate URLs.

Comment (jdolitsky) [reply in_toto/model.go:83]: LGTM

**Comment by SantiagoTorres** [comment]
Bump here, I think things seem reasonable? @puerco @jdolitsky agreed?

**Comment by jdolitsky** [comment]
lgtm!

**Comment by SantiagoTorres** [comment]
Merging! sorry to the long RTT!

**Comment by masahiro331** [comment]
@puerco @jdolitsky 

Thank you all for a lovely product!
I have one question.

"https://cyclonedx.org/schema" does not seem to be the URL of the document.
For example, this non-existent URL refers to the same page.
> https://cyclonedx.org/schemaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

This URL may refer to the correct schema.
https://cyclonedx.org/docs/1.4/json/

However, the challenge is that currently, it is not possible to identify the version unless cyclonedx is parsed.

I consider the correct URL should be set as a specification.
I want to discuss this problem.

Apologies if this is out of place.

**Pull request merged**